web/src/app/loading/components/Spinner.js

```diff
@@ -8,66 +8,62 @@ import { DEFAULT_SPIN_DELAY_MS, DEFAULT_SPIN_WAIT_MS } from '../../config'
 /*
  * Show a loading spinner in the center of the container.
  */
-export default class Spinner extends React.PureComponent {
-  static propTypes = {
-    // Wait `delayMs` milliseconds before rendering a spinner.
-    delayMs: p.number,
+export default function Spinner(props) {
+  const [spin, setSpin] = useState(false)
 
-    // Wait `waitMs` before calling onReady.
-    waitMs: p.number,
+  useEffect(() => {
+    let _spin = setTimeout(() => {
+      _spin = null
+      setSpin(true)
+      if (props.onSpin) props.onSpin()
 
-    // onSpin is called when the spinner starts spinning.
-    onSpin: p.func,
+      if (props.waitMs && props.onReady) {
+        _spin = setTimeout(props.onReady, props.waitMs)
+      }
+    }, props.delayMs)
 
-    // onReady is called once the spinner has spun for `waitMs`.
-    onReady: p.func,
+    return () => {
+      clearTimeout(_spin)
+    }
+  })
```

Comment:
```suggestion
  }, [])
```

You'll want to add an empty array as the second argument to `useEffect` so that it only runs once to mimic `componentDidMount`